src/oneD/StFlow.cpp

```diff
@@ -1014,17 +1025,14 @@ void StFlow::evalRightBoundary(double* x, double* rsd, int* diag, double rdt)
     }
     rsd[index(c_offset_Y + rightExcessSpecies(), j)] = 1.0 - sum;
     diag[index(c_offset_Y + rightExcessSpecies(), j)] = 0;
-    if (!m_isFree) {
-        rsd[index(c_offset_U,j)] = rho_u(x,j);
-        if (m_do_energy[j]) {
-            rsd[index(c_offset_T,j)] = T(x,j);
-        } else {
-            rsd[index(c_offset_T, j)] = T(x,j) - T_fixed(j);
-        }
+    if (m_usesLambda) {
+        rsd[index(c_offset_U, j)] = rho_u(x, j);
     } else {
-        rsd[index(c_offset_U,j)] = rho_u(x,j) - rho_u(x,j-1);
-        rsd[index(c_offset_T,j)] = T(x,j) - T(x,j-1);
+        rsd[index(c_offset_U, j)] = rho_u(x, j) - rho_u(x, j-1);
     }
+    rsd[index(c_offset_L, j)] = lambda(x, j);
```

Comment:
In the counterflow case at least, I think this right boundary condition should be `lambda(x, j) - lambda(x, j-1)`. A right `Inlet1D` object does not modify this value, and this results in $\Lambda = 0$ at the right, when it should have the same constant value as at all other grid points.

Reply:
Good catch.